created full stop and peat (#1747)
urbit/urbit.org: content/grants/full-stop.md

At the risk of sounding flippant, Urbit fixes a great deal of this for users without interrupting their computing experience. Urbit provides an encrypted communication layer with identity primatives (affording identity assurances in comms). Urbit also acts as a personal server, avoiding the concern of corporate leaks of biometric data entirely. And yet, Urbit is available on a user's Phone as an interface for inputting data, maintaining the convenience of otherwise-centralized services. A period tracker on Urbit is simple solution to a very complex problem. It also presents some interesting development questions, reviewed below.

# Overview
Full Stop helps you track your period and period-related-accessory-phenomena on a secure, private, personal server called an urbit. **Full Stop** is accessed like a traditional phone application, available on major phone operating systems. **Full Stop** will include basic Push Notifications which will, given the need to pass through Apple's Great Filter, never include private information and should provide no indicia of a user's health conditions (or lack thereof) in their frequency or rate of occurrence.

> On first opening **Full Stop**, a user should enter a simple PIN, in addition to their `+code` required to log into their Urbit, generally. This pin provides a final layer of security, on top of their Phone's security, on top of Urbit's encryption and login requirements, as an additional precaution to ensure private access, only.

Reviewer: ```suggestion
**Full Stop** should provide a variety of secure methods for accessing and inputting data, including a web-app, asynchronous apps built by core devs and 3rd parties, etc. If users don't trust the in-the-box web-app, they should be free to construct their own.
```

> After establishing their security PIN, Full Stop will proceed as with most period tracker applications:

Reviewer: ```suggestion
Through whatever front-end the user chooses, Full Stop's available functions will track with most period tracker applications currently on the market:
```

### Recording Data
- **Period Data:** The user can record information about their period.
    - Start and Stop dates
    - Flow rate per day, during period

>     - Pain and Physical Symptom checklist per day

Reviewer: ```suggestion
    - Physical Symptom checklist per day
```

- **Sexual Activity Data:** The user can record information about any sexual contact they may experience.
    - Date of encounter
    - Use or non-use of additional contraceptive protection (e.g. condoms)
    - User's partner's @p (kidding)
- **Additional BioMarkers:** The user can record period-adjacent data they may wish to record.
    - Off-period spotting

>     - Birth Control Pill consumption tracker (possibly alerts, though we're considering the data leak possibilities through push notifications)

Reviewer: ```suggestion
    - Birth Control Pill consumption tracker
    - Term Birth Control application and renewal date tracker (e.g. IUD)
 - **Pregnancy Tracking:** The user can record information about their pregnancy, and track the weeks until their due date.
   - User can set a due date from the date of the start of the pregnancy or their projected due date.
   - The application will provide the user with a count of weeks since the start of their pregnancy.
- **Fertility Prediction:** The user can elect to view various projections of fertility windows based on their prior period data, as an aid for con(tra)ception.
   - Methods include timing methods and, later, application of basal body temp and cervical mucosal density data inputted by the user to further specify fertility windows 
```

### Reviewing Data
- **Data Visualization:** Visualize cycles over time, along with predictive models of likely windows of fertility, hormonal activity (based on current medical literature), and the likely window of the user's next period.
- **Orientation Assistance:** Should a user choose, Full Stop will assist them in planning sexual encounters to most likely result or not result in a pregnancy
    - **Full Stop** will incorporate several methods of timing contraception users can select from including those proscribed by various women's health organizations as potential alternatives to hormonal or physical birth control options (the 'Rhythm Method', Natural Family Planning, the Standard Day method).
    - **Full Stop** will also incorporate available ovulation/fertility window timing algorithms users may select from, to assist them in becoming pregnant.

> ### Outputting Data
> - **U on Urbit:**
>     - **U on Urbit** is a Biometric Data Suite for Urbit. Users of **U** will be able to track a wide array of their personal biometric data on their personal server.
>     - Rather than attempting to build a comprehensive, monolithic health application, **U** combines many, separate biometric input streams that correspond with the way Users already use other devices/applications to track, e.g. calories, steps, heart rate, recurring medication, insulin levels. Any development team should be able to add a new application or device, whos data **U** should be able to injest.

Reviewer: ```suggestion
### Outputting Data
- Application data should be available for use in other applications, pending inter-app-permissions development.
```

>     - The **U on Urbit** product lineup will begin with **Full Stop**.

Reviewer: ```suggestion

```

## User Stories
As a user, I can:
- Download an iOS/Android App from the app store,

> - Point their phone app at an Urbit (by IP or URL) and enter their `+code` to connect the app to their Urbit,

Reviewer: ```suggestion
- Point my phone app at an Urbit (by IP or URL) and enter my `+code` to connect the app to their Urbit,
```

> - Optionally set an additional PIN in the app to control access,

Reviewer: ```suggestion

```

> - Track my period within the application including:
>     - Start Date,
>     - Stop Date,
>     - Flow Rate per Day,
>     - Physical Symptoms (pain, cramping, clotting, etc.),
> - Track my sexual activity within the application including:
>     -Sexual Contact Date,
>     -Use/Non-Use of Prophylactic,
> - Track additional biomarkers related to menstruation:
>     - Off-Period Spotting,
>     - Birth control pill consumption,

Reviewer: ```suggestion
- Track my period within the application including:
    - Start Date,
    - Stop Date,
    - Flow Rate per Day,
    - Physical Symptoms (pain, cramping, clotting, etc.),
- Track my sexual activity within the application including:
    -Sexual Contact Date,
    -Use/Non-Use of Prophylactic,
- Track additional biomarkers related to menstruation:
    - Off-Period Spotting,
    - Birth control pill consumption,
- Track the weeks of my pregnancy
- Project my fertile days based on my own data stored on my secured server
```
